### PR TITLE
Assess ip_addresses by comparing the host and network octets

### PR DESCRIPTION
**Pull request by JonathanAndrews**
This PR changes how we compare IP addresses.
1. Check the IP addresses aren't an exact match --> if they are, we can return and award 3 points.
2. Check the IP addresses' Network Octets aren't a match --> if they are, we can return and award 2 points.
3. Check the IP addresses are within the `max_ip_distance` of each other --> if they are, we can return and award 1 point.
4. If the IP addresses are completely different, we award no points.

Re: the `#failures` debug method:
Unless the IP addresses are an exact match, we log the IP addresses and the `comparison_result` to Rollbar.

### CODE REVIEW

**Comment by jalada** [comment]
Thx @JonathanAndrews on my list to review today!

**Comment by jalada** [comment]
@JonathanAndrews we should have a simple fallback for when the IP address is IPv6 instead of IPv4. Networks etc. don't work in the same way with IPv6, so we should only do an exact match in that situation.

File: app/lib/devise_recognisable/guard.rb [hunk 1]
```diff
@@ -61,19 +65,33 @@ def calculate_score_for(session)
...
   def compare_ip_addresses(session_address)
-    return true if session_address == @request.location.ip
-    
+    # Check if the IP is an exact match
+    return :exact_match if session_address == @request.location.ip
+
+    # Check that neither IP address is IPv6.
+    # If either is, return :ip_version_mismatch.
+    if ipv6?(@request.location.ip) || ipv6?(session_address)
+      return :ip_version_mismatch
```
Comment (jalada) [inline app/lib/devise_recognisable/guard.rb:76]: It doesn't quite feel like this is the right description. If both the incoming request _and_ session IP were IPv6 we'd say 'there's a version mismatch' but that's not true. Instead I'd just return `:complete_mismatch` as that's all we're going to do for IPv6 (e.g. either it's an exact match or nothing).

Comment (JonathanAndrews) [reply app/lib/devise_recognisable/guard.rb:76]: 👍  Yep

**Pull request merged**